src/Compiler/TypedTree/TypedTree.fs

```diff
@@ -2260,6 +2260,9 @@ type TyparOptionalData =
 
       /// Set to true if the typar is contravariant, i.e. declared as <in T> in C#
       mutable typar_is_contravariant: bool
+
+      /// The declared name of the type parameter.
+      mutable typar_declared_name: string option
```

Comment:
How does it make its way into the generated IL ?

Reply:
It does not, that's the point :) 

The original issue was https://github.com/dotnet/fsharp/issues/15286 and the original fix for it was to overwrite the incoming typar idents with the ones from extensions (#15287) . This PR kind of reverts to standard behavior in this regard, so the IL for type extensions contains the type parameters of the extended type.